base_field_validator/models/ir_model.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    See __openerp__.py about license
+#
+##############################################################################
+
+import re
+from openerp.tools.translate import _
+import openerp
+from openerp import SUPERUSER_ID
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+from openerp.osv import orm
+
+
+class IrModel(models.Model):
+
+    _inherit = 'ir.model'
+    validator_line_ids = fields.One2many(
+        'ir.model.validator.line', 'name', 'Validators')
+
+    @api.model
+    def check_vals(self, vals, model_name):
+        validator_lines = self.env['ir.model.validator.line'].search([
+            ('name.model', '=', model_name),
+            ('field_id.name', 'in', vals.keys())])
+        for validator_line in validator_lines:
+            if validator_line.field_id.name in vals:
```

Comment:
This check is now redundant, as we already filter on the field name in the search call.